packages/synapse-react-client/src/synapse-queries/subscription/useSubscription.ts

```diff
@@ -68,16 +138,24 @@ export function usePostSubscription(
     {
       ...options,
       onSuccess: async (updatedSubscription, variables, ctx) => {
-        await queryClient.invalidateQueries(
-          keyFactory.getAllSubscriptionsQueryKey(),
-        )
-
-        await queryClient.invalidateQueries(
-          keyFactory.getSubscribersQueryKey(
-            variables.objectId,
-            variables.objectType,
+        await Promise.all([
+          queryClient.invalidateQueries(
+            keyFactory.getAllSubscriptionsQueryKey(),
+          ),
+          queryClient.invalidateQueries(
+            keyFactory.getSubscriptionQueryKey(
+              variables.objectId,
+              variables.objectType,
+            ),
           ),
-        )
+          queryClient.invalidateQueries(
+            keyFactory.getSubscribersQueryKey(
+              variables.objectId,
+              variables.objectType,
+            ),
+          ),
+        ])
```

Comment:
This is now more specific. Before it would invalidate all subscription-related queries. Now it invalidates the subscription list query, and the subscription queries related just to this object (so now, subscribing to Forum A doesn't invalidate the cache for `useGetSubscription` for Forum B)